src/test/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilderMetricsLabelTest.java

```diff
@@ -0,0 +1,235 @@
+              assertNotNull(labels);
+              assertNotNull(labels.get(KubernetesEngineBuilder.METRICS_LABEL_KEY));
+              assertEquals(
+                  labels.get(KubernetesEngineBuilder.METRICS_LABEL_KEY),
+                  KubernetesEngineBuilder.METRICS_LABEL_VALUE);
+              return null;
+            })
+        .when(manifestFile)
+        .write(anyString(), anyString());
+    KubernetesEngineBuilder.addMetricsLabel(manifestFile);
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testAddMetricsLabelProperlyHandlesExistingMetricsLabel()
+      throws IOException, InterruptedException {
+    FilePath manifestFile = Mockito.mock(FilePath.class);
+    Mockito.when(manifestFile.read())
+        .thenReturn(
+            new ByteArrayInputStream(
+                String.join(
+                        "\n",
+                        "apiVersion: apps/v1",
+                        "kind: Service",
+                        "metadata:",
+                        "  name: nginx-service",
+                        "  app.kubernetes.io/managed-by: graphite-jenkins-gke")
+                    .getBytes()));
+
+    Mockito.doAnswer(
+            invocation -> {
+              Yaml yaml = new Yaml();
+              Manifests.ManifestObject manifest =
+                  new Manifests.ManifestObject(
+                      (Map<String, Object>) yaml.load((String) invocation.getArguments()[0]),
+                      manifestFile);
+              Map<String, String> labels = manifest.getOrCreateLabels();
+              assertNotNull(labels);
+              assertNotNull(labels.get(KubernetesEngineBuilder.METRICS_LABEL_KEY));
+              assertEquals(
+                  labels.get(KubernetesEngineBuilder.METRICS_LABEL_KEY),
+                  KubernetesEngineBuilder.METRICS_LABEL_VALUE);
+              return null;
+            })
+        .when(manifestFile)
+        .write(anyString(), anyString());
+    KubernetesEngineBuilder.addMetricsLabel(manifestFile);
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testAddMetricsLabelProperlyHandlesMultipleObjects()
+      throws IOException, InterruptedException {
+    FilePath manifestFile = Mockito.mock(FilePath.class);
+    Mockito.when(manifestFile.read())
+        .thenReturn(
+            new ByteArrayInputStream(
+                String.join(
+                        "\n",
+                        "apiVersion: apps/v1",
+                        "kind: Service",
+                        "metadata:",
+                        "  name: nginx-service",
+                        "\n",
+                        "---",
+                        "apiVersion: apps/v1",
+                        "kind: Deployment",
+                        "metadata:",
+                        "  name: nginx-deployment",
+                        "\n",
+                        "---",
+                        "apiVersion: apps/v1",
+                        "kind: ReplicaSet",
```

Comment:
Suggest adding a test that defines an object with a type that isn't in METRICS_TARGET_TYPES in the manifest and verifies that the label is not applied.

Reply:
Done.